app/controllers/CommodityCodeController.scala

```diff
@@ -107,14 +107,20 @@ class CommodityCodeController @Inject() (
         .fold(
           formWithErrors => Future.successful(BadRequest(view(formWithErrors, onSubmitAction))),
           value =>
-            (for {
-              commodity               <-
-                ottConnector.getCommodityCode(value, request.eori, request.affinityGroup, CreateRecordJourney, None)
-              updatedAnswers          <- Future.fromTry(request.userAnswers.set(CommodityCodeUpdatePage(recordId), value))
-              updatedAnswersWithQuery <-
-                Future.fromTry(updatedAnswers.set(CommodityUpdateQuery(recordId), commodity))
-              _                       <- sessionRepository.set(updatedAnswersWithQuery)
-            } yield Redirect(navigator.nextPage(CommodityCodeUpdatePage(recordId), mode, updatedAnswersWithQuery)))
+            {
+              val oldValue = request.userAnswers.get(CommodityCodeUpdatePage(recordId)).getOrElse("")
```

Comment:
Is it isn't set it will return an empty option, we wouldn't want to default it to an empty string here but instead map on the option and check if it matches

Reply:
not defaulting to string and using option now